src/Microsoft.AspNetCore.ApplicationInsights.HostingStartup/JavaScriptSnippetTagHelperComponent.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.ApplicationInsights.AspNetCore;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+
+namespace Microsoft.AspNetCore.ApplicationInsights.HostingStartup
+{
+    /// <summary>
+    /// The <see cref="TagHelperComponent"/> that injects the <see cref="JavaScriptSnippet"/> to the head tag.
+    /// </summary>
+    public class JavaScriptSnippetTagHelperComponent : TagHelperComponent
```

Comment:
Why is this not a good name? We usually suffix types with their base type (assuming it's of some interest, which in this case, I believe it is). Also, I think no one will ever see this type name.